### PR TITLE
[core] Lower current scheduler priority if SCHED_IDLE is not supported

### LINKED ISSUE #5535: Set thread priority on Android
Currently, we're not setting the thread priority in Android: https://github.com/mapbox/mapbox-gl-native/blob/da863c6e52f656bd35c3d3346093a24d747d0bbd/platform/default/thread.cpp#L23

It seems possible to set the value, according to http://www.androiddesignpatterns.com/2014/01/thread-scheduling-in-android.html, but I'm not sure if we need to do it from Java, or if there's a way to call the underlying API from C++.

/cc @bleege @ivovandongen @tobrun

### PR DESCRIPTION
**Pull request by tmpsantos**
Still reduces scheduler priority for Android NDK API lower than 21 where SCHED_IDLE is not defined.

Fixes #5535

### CODE REVIEW

**Comment by tmpsantos** [comment]
:eyes: @kkaefer

**Comment by tmpsantos** [comment]
Hmm, on Android this doesn't work. Not even if we force the scheduler type. I will see how Chromium is doing and work something on the same lines.

https://chromium.googlesource.com/chromium/src/base/+/ababb4cf8b6049a642a2f361b1006a07561c2d96/threading/platform_thread_android.cc

**Comment by tmpsantos** [comment]
This now works and it is based on how Google Chrome is doing. Our threads have the priority 139 which is the lowest on the process group of the example app.

```
prio: 116      name: RenderThread
prio: 118      name: hwuiTask1
prio: 120      name: Alarm
prio: 120      name: Binder_1
prio: 120      name: Binder_2
prio: 120      name: Binder_3
prio: 120      name: FinalizerDaemon
prio: 120      name: FinalizerWatchd
prio: 120      name: HeapTaskDaemon
prio: 120      name: JDWP
prio: 120      name: LeakCanary-File
prio: 120      name: LeakCanary-Heap
prio: 120      name: ReferenceQueueD
prio: 120      name: Signal Catcher
prio: 120      name: Timer-0
prio: 120      name: pboxsdk.testapp
prio: 120      name: pboxsdk.testapp
prio: 120      name: pboxsdk.testapp
prio: 120      name: pboxsdk.testapp
prio: 120      name: pboxsdk.testapp
prio: 120      name: pboxsdk.testapp
prio: 120      name: pboxsdk.testapp
prio: 120      name: pool-1-thread-1
prio: 130      name: AsyncTask #1
prio: 130      name: AsyncTask #2
prio: 130      name: AsyncTask #3
prio: 130      name: AsyncTask #4
prio: 130      name: AsyncTask #5
prio: 130      name: AsyncTask #6
prio: 139      name: AssetFileSource
prio: 139      name: DefaultFileSour
prio: 139      name: Okio Watchdog
prio: 139      name: Worker
prio: 139      name: Worker
prio: 139      name: Worker
prio: 139      name: Worker
```

**Comment by tmpsantos** [comment]
:eyes: @bleege @zugaldia @tobrun

File: platform/android/src/thread.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include <mbgl/platform/log.hpp>
+#include <mbgl/platform/platform.hpp>
+
+#include <sys/prctl.h>
+#include <sys/resource.h>
+
+// Implementation based on Chromium's platform_thread_android.cc.
```
Comment (zugaldia) [inline platform/android/src/thread.cpp]: Nice :)

**Pull request merged**